### DISCUSSION TITLE
tutorials/navigation/navigation_introduction_3d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#157] [User-contributed notes]
# tutorials/navigation/navigation_introduction_3d

Godot provides multiple objects, classes and servers to facilitate grid-based or mesh-based navigation and pathfinding for 2D and 3D games. The following section provides a quick overview over all ...

https://docs.godotengine.org/en/stable/tutorials/navigation/navigation_introduction_3d.html

**Comment by jak6jak** [comment]
If your agent isn't moving it could possibly be because it's path makes it collide inside the ground. To fix this change the path height offset. For a basic default capsule collision shape I changed mine to -1 and it worked.

**Reply by rymkapro** [reply]
Im sorry, im new in Godot. Is it possible calculate path height offset somehow, because i changed it no negative value and agent start really moving, but it is not on the ground now :)

**Comment by nerveship** [comment]
If you're having issues with an agent not moving (in my case, it was being larger than the default capsule size), you may want to try this. 

Create a Node3D within your scene and place it at the bottom of your collision shape, add the NavigationAgent3D node as a child of that. This saved me from having to mess with the path_height_offset, and also solved my problems with slopes.
